src/components/editor-page/document-bar/aliases/aliases-add-form.test.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { render, act, screen } from '@testing-library/react'
+import testEvent from '@testing-library/user-event'
+import React from 'react'
+import { mockI18n } from '../../../markdown-renderer/test-utils/mock-i18n'
+import * as AliasModule from '../../../../api/alias'
+import * as NoteDetailsReduxModule from '../../../../redux/note-details/methods'
+import * as useApplicationStateModule from '../../../../hooks/common/use-application-state'
+import { AliasesAddForm } from './aliases-add-form'
+import * as useUiNotificationsModule from '../../../notifications/ui-notification-boundary'
+
+jest.mock('../../../../api/alias')
+jest.mock('../../../../redux/note-details/methods')
+jest.mock('../../../../hooks/common/use-application-state')
+jest.mock('../../../notifications/ui-notification-boundary')
+
+const addPromise = Promise.resolve({ name: 'mock', primaryAlias: true, noteId: 'mock' })
+
+describe('AliasesAddForm', () => {
+  beforeEach(async () => {
+    await mockI18n()
+    jest.spyOn(AliasModule, 'addAlias').mockImplementation(() => addPromise)
+    jest.spyOn(NoteDetailsReduxModule, 'updateMetadata').mockImplementation(() => Promise.resolve())
+    jest.spyOn(useApplicationStateModule, 'useApplicationState').mockReturnValue('mock-note')
+    jest.spyOn(useUiNotificationsModule, 'useUiNotifications').mockReturnValue({
+      showErrorNotification: jest.fn(),
+      dismissNotification: jest.fn(),
+      dispatchUiNotification: jest.fn()
+    })
+  })
+
+  afterAll(() => {
+    jest.resetAllMocks()
+    jest.resetModules()
+  })
+
+  it('renders the input form', async () => {
+    const view = render(<AliasesAddForm />)
+    expect(view.container).toMatchSnapshot()
+    const button = screen.getByTitle('editor.modal.aliases.addAlias')
```

Comment:
Use `findByTestId` or `getByTestId` instead. Makes the purpose more independent from the content.
You can use `testId('id')` to add this attribute to dom elements.
See e.g. https://github.com/hedgedoc/react-client/blob/f2ed9d44531fe64c4ea0a1db191344f0636fbfa1/src/components/common/note-loading-boundary/create-non-existing-note-hint.tsx#L50

Reply:
I also prefer to use `find` methods over `get` because `find` waits until the element appears while `get` checks only the current state. Helpful if you're waiting for certain change to happen e.g. because of an effect.

Reply:
Changed